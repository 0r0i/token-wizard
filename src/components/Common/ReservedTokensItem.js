import React from 'react'

const ReservedTokensItem = props => {
  const deleteButton = props.readOnly ? null : <div onClick={() => props.onRemove(props.num)} />

  return (
    <div className="sw-ReservedTokensItem">
      <div className="sw-ReservedTokensItem_Column">{props.addr}</div>
      <div className="sw-ReservedTokensItem_Column">{props.dim}</div>
      <div className="sw-ReservedTokensItem_Column">{props.val}</div>
      {deleteButton}
    </div>
  )
}

export default ReservedTokensItem
