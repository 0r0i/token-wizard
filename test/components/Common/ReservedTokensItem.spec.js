import React from 'react'
import ReservedTokensItem from '../../../src/components/Common/ReservedTokensItem'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'

configure({ adapter: new Adapter() })

const token = {
  addr: '0x1111111111111111111111111111111111111111',
  dim: 'percentage',
  val: '12.1234'
}

describe('ReservedTokensItem', () => {
  // TODO: modify this test after component refactor
  // it(`should render the component with delete button enabled`, () => {
  //   const wrapper = mount(
  //     shallow(
  //       <ReservedTokensItem key={0} num={0} addr={token.addr} dim={token.dim} val={token.val} readOnly={false} />
  //     ).get(0)
  //   )
  //
  //   expect(wrapper).toMatchSnapshot()
  //   expect(wrapper.find('.reserved-tokens-item-empty')).toHaveLength(1)
  // })

  it(`should render the component with no delete button`, () => {
    const wrapper = mount(
      shallow(
        <ReservedTokensItem key={0} num={0} addr={token.addr} dim={token.dim} val={token.val} readOnly={true} />
      ).get(0)
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('reserved-tokens-item-empty')).toHaveLength(0)
  })

  // TODO: modify this test after component refactor
  // it(`should receive the item index on remove button click`, () => {
  //   const onRemove = jest.fn()
  //   const wrapper = mount(
  //     <ReservedTokensItem
  //       key={0}
  //       num={0}
  //       addr={token.addr}
  //       dim={token.dim}
  //       val={token.val}
  //       readOnly={false}
  //       onRemove={onRemove}
  //     />
  //   )
  //
  //   wrapper.find('a').simulate('click')
  //
  //   expect(onRemove).toHaveBeenCalledTimes(1)
  //   expect(onRemove).toHaveBeenLastCalledWith(0)
  // })
})
