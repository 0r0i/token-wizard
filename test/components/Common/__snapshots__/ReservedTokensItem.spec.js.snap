// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedTokensItem should render the component with delete button enabled 1`] = `
<div
  className="sw-ReservedTokensItem"
>
  <div
    className="sw-ReservedTokensItem_Column"
  >
    0x1111111111111111111111111111111111111111
  </div>
  <div
    className="sw-ReservedTokensItem_Column"
  >
    percentage
  </div>
  <div
    className="sw-ReservedTokensItem_Column"
  >
    12.1234
  </div>
  <div
    onClick={[Function]}
  />
</div>
`;

exports[`ReservedTokensItem should render the component with no delete button 1`] = `
<div
  className="sw-ReservedTokensItem"
>
  <div
    className="sw-ReservedTokensItem_Column"
  >
    0x1111111111111111111111111111111111111111
  </div>
  <div
    className="sw-ReservedTokensItem_Column"
  >
    percentage
  </div>
  <div
    className="sw-ReservedTokensItem_Column"
  >
    12.1234
  </div>
</div>
`;
